### ISSUE TITLE
Training Successful... But not really??

### ISSUE THREAD

**Issue by WolfLink115** [LarryJane491/Lora-Training-in-Comfy#107]
So, I am using an AMD system in Windows. I did a whole lot of changes with both ComfyUI and the scripts to accommodate for having an AMD card. The thing is, it claims it finishes successfully, and generates a safetensors file perfectly fine, however the avr_loss is always 0, and according to Google Gemini (I know, I tried asking an AI for help with AI, bite me lol) it claims that if the avr_loss is 0 the whole time, then the AI isn't learning anything. I only have 4 images in the a folder, and it is named correctly afaik (4_lpf). I don't know if it is working or failing. Please help-

Here is the whole log:

`G:\AIGeneration\ComfyUI_windows_portable>set MIOPEN_FIND_MODE=2

G:\AIGeneration\ComfyUI_windows_portable>set MIGRAPHX_MLIR_USE_SPECIFIC_OPS="attention"

G:\AIGeneration\ComfyUI_windows_portable>set MIOPEN_LOG_LEVEL=3

G:\AIGeneration\ComfyUI_windows_portable>set TORCH_ROCM_AOTRITON_ENABLE_EXPERIMENTAL=1

G:\AIGeneration\ComfyUI_windows_portable>set XFORMERS_FORCE_DISABLE=1

G:\AIGeneration\ComfyUI_windows_portable>set HSA_OVERRIDE_GFX_VERSION=11.0.0

G:\AIGeneration\ComfyUI_windows_portable>set PYTORCH_HIP_ALLOC_CONF=expandable_segments:True

G:\AIGeneration\ComfyUI_windows_portable>.\python_embeded\python.exe -s ComfyUI\main.py --windows-standalone-build --disable-smart-memory --async-offload --use-pytorch-cross-attention --enable-manager --fp32-vae
Checkpoint files will always be loaded safely.
Total VRAM 16368 MB, total RAM 51180 MB
pytorch version: 2.10.0a0+rocm7.10.0a20251120
Set: torch.backends.cudnn.enabled = False for better AMD performance.
AMD arch: gfx1101
ROCm version: (7, 10)
Set vram state to: NORMAL_VRAM
Disabling smart memory management
Device: cuda:0 AMD Radeon RX 7800 XT : native
Using async weight offloading with 2 streams
Enabled pinned memory 23031.0
Using pytorch attention
Python version: 3.12.10 (tags/v3.12.10:0cc8128, Apr  8 2025, 12:21:36) [MSC v.1943 64 bit (AMD64)]
ComfyUI version: 0.4.0
ComfyUI frontend version: 1.34.8
[Prompt Server] web root: G:\AIGeneration\ComfyUI_windows_portable\python_embeded\Lib\site-packages\comfyui_frontend_package\static
Total VRAM 16368 MB, total RAM 51180 MB
pytorch version: 2.10.0a0+rocm7.10.0a20251120
Set: torch.backends.cudnn.enabled = False for better AMD performance.
AMD arch: gfx1101
ROCm version: (7, 10)
Set vram state to: NORMAL_VRAM
Disabling smart memory management
Device: cuda:0 AMD Radeon RX 7800 XT : native
Using async weight offloading with 2 streams
Enabled pinned memory 23031.0
(pysssss:WD14Tagger) [DEBUG] Available ORT providers: AzureExecutionProvider, CPUExecutionProvider
(pysssss:WD14Tagger) [DEBUG] Using ORT providers: CUDAExecutionProvider, CPUExecutionProvider

Import times for custom nodes:
   0.0 seconds: G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\custom_nodes\websocket_image_save.py
   0.0 seconds: G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\custom_nodes\Image-Captioning-in-ComfyUI
   0.0 seconds: G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\custom_nodes\Lora-Training-in-Comfy
   0.0 seconds: G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-WD14-Tagger
   0.0 seconds: G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Custom-Scripts

Context impl SQLiteImpl.
Will assume non-transactional DDL.
No target revision found.
Starting server

To see the GUI go to: http://127.0.0.1:8188
[DEPRECATION WARNING] Detected import of deprecated legacy API: /scripts/ui.js. This is likely caused by a custom node extension using outdated APIs. Please update your extensions or contact the extension author for an updated version.
[DEPRECATION WARNING] Detected import of deprecated legacy API: /scripts/ui/components/button.js. This is likely caused by a custom node extension using outdated APIs. Please update your extensions or contact the extension author for an updated version.
got prompt
[]
G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\custom_nodes\Lora-Training-in-Comfy/sd-scripts/train_network.py
prepare tokenizer
update token length: 225
Using DreamBooth method.
prepare images.
found directory G:\AIGeneration\LoraTraining\4_lpf contains 4 image files
16 train images with repeating.
0 reg images.
no regularization images / 正則化画像が見つかりませんでした
[Dataset 0]
  batch_size: 1
  resolution: (512, 512)
  enable_bucket: True
  min_bucket_reso: 256
  max_bucket_reso: 1584
  bucket_reso_steps: 64
  bucket_no_upscale: False

  [Subset 0 of Dataset 0]
    image_dir: "G:\AIGeneration\LoraTraining\4_lpf"
    image_count: 4
    num_repeats: 4
    shuffle_caption: True
    keep_tokens: 0
    caption_dropout_rate: 0.0
    caption_dropout_every_n_epoches: 0
    caption_tag_dropout_rate: 0.0
    caption_prefix: None
    caption_suffix: None
    color_aug: False
    flip_aug: False
    face_crop_aug_range: None
    random_crop: False
    token_warmup_min: 1,
    token_warmup_step: 0,
    is_reg: False
    class_tokens: lpf
    caption_extension: .txt

[Dataset 0]
loading image sizes.
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 622.51it/s]
make buckets
number of images (including repeats) / 各bucketの画像枚数（繰り返し回数を含む）
bucket 0: resolution (448, 576), count: 8
bucket 1: resolution (576, 448), count: 8
mean ar error (without repeats): 0.10007134140930243
preparing accelerator
C:\Users\wolfl\AppData\Roaming\Python\Python313\site-packages\accelerate\accelerator.py:529: UserWarning: `log_with=tensorboard` was passed but no supported trackers are currently installed.
  warnings.warn(f"`log_with={log_with}` was passed but no supported trackers are currently installed.")
loading model for process 0/1
load StableDiffusion checkpoint: G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\models\checkpoints\v1-5-pruned.safetensors
UNet2DConditionModel: 64, 8, 768, False, False
loading u-net: <All keys matched successfully>
loading vae: <All keys matched successfully>
loading text encoder: _IncompatibleKeys(missing_keys=[], unexpected_keys=['text_model.embeddings.position_ids'])
Enable memory efficient attention for U-Net
import network module: networks.lora
C:\Users\wolfl\AppData\Roaming\Python\Python313\site-packages\torch\nn\modules\module.py:1369: UserWarning: expandable_segments not supported on this platform (Triggered internally at B:\src\torch\c10/hip/HIPAllocatorConfig.h:40.)
  return t.to(
[Dataset 0]
caching latents.
checking cache validity...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 155344.59it/s]
caching latents...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00,  5.11it/s]
create LoRA network. base dim (rank): 32, alpha: 32.0
neuron dropout: p=None, rank dropout: p=None, module dropout: p=None
create LoRA for Text Encoder:
create LoRA for Text Encoder: 72 modules.
create LoRA for U-Net: 192 modules.
enable LoRA for text encoder
enable LoRA for U-Net
prepare optimizer, data loader etc.
use AdamW optimizer | {}
override steps. steps for 50 epochs is / 指定エポックまでのステップ数: 800
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 16
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 16
  num epochs / epoch数: 50
  batch size per device / バッチサイズ: 1
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 800
steps:   0%|                                                                                                                                                                            | 0/800 [00:00<?, ?it/s]
epoch 1/50
steps:   2%|███                                                                                                                                                    | 16/800 [00:05<04:40,  2.80it/s, avr_loss=0]
epoch 2/50
steps:   4%|██████                                                                                                                                                 | 32/800 [00:09<03:52,  3.30it/s, avr_loss=0]
epoch 3/50
steps:   6%|█████████                                                                                                                                              | 48/800 [00:13<03:34,  3.50it/s, avr_loss=0]
epoch 4/50
steps:   8%|████████████                                                                                                                                           | 64/800 [00:17<03:23,  3.61it/s, avr_loss=0]
epoch 5/50
steps:  10%|███████████████                                                                                                                                        | 80/800 [00:21<03:15,  3.68it/s, avr_loss=0]
epoch 6/50
steps:  12%|██████████████████                                                                                                                                     | 96/800 [00:25<03:08,  3.73it/s, avr_loss=0]
epoch 7/50
steps:  14%|█████████████████████                                                                                                                                 | 112/800 [00:29<03:02,  3.77it/s, avr_loss=0]
epoch 8/50
steps:  16%|████████████████████████                                                                                                                              | 128/800 [00:33<02:57,  3.79it/s, avr_loss=0]
epoch 9/50
steps:  18%|███████████████████████████                                                                                                                           | 144/800 [00:37<02:52,  3.81it/s, avr_loss=0]
epoch 10/50
steps:  20%|██████████████████████████████                                                                                                                        | 160/800 [00:42<02:48,  3.81it/s, avr_loss=0]
saving checkpoint: G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\models\loras\lpf-000010.safetensors

epoch 11/50
steps:  22%|█████████████████████████████████                                                                                                                     | 176/800 [00:46<02:44,  3.79it/s, avr_loss=0]
epoch 12/50
steps:  24%|████████████████████████████████████                                                                                                                  | 192/800 [00:50<02:39,  3.80it/s, avr_loss=0]
epoch 13/50
steps:  26%|███████████████████████████████████████                                                                                                               | 208/800 [00:54<02:35,  3.81it/s, avr_loss=0]
epoch 14/50
steps:  28%|██████████████████████████████████████████                                                                                                            | 224/800 [00:58<02:30,  3.82it/s, avr_loss=0]
epoch 15/50
steps:  30%|█████████████████████████████████████████████                                                                                                         | 240/800 [01:02<02:26,  3.83it/s, avr_loss=0]
epoch 16/50
steps:  32%|████████████████████████████████████████████████                                                                                                      | 256/800 [01:06<02:21,  3.84it/s, avr_loss=0]
epoch 17/50
steps:  34%|███████████████████████████████████████████████████                                                                                                   | 272/800 [01:10<02:17,  3.85it/s, avr_loss=0]
epoch 18/50
steps:  36%|██████████████████████████████████████████████████████                                                                                                | 288/800 [01:14<02:12,  3.86it/s, avr_loss=0]
epoch 19/50
steps:  38%|█████████████████████████████████████████████████████████                                                                                             | 304/800 [01:18<02:08,  3.86it/s, avr_loss=0]
epoch 20/50
steps:  40%|████████████████████████████████████████████████████████████                                                                                          | 320/800 [01:22<02:04,  3.87it/s, avr_loss=0]
saving checkpoint: G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\models\loras\lpf-000020.safetensors

epoch 21/50
steps:  42%|███████████████████████████████████████████████████████████████                                                                                       | 336/800 [01:27<02:00,  3.85it/s, avr_loss=0]
epoch 22/50
steps:  44%|██████████████████████████████████████████████████████████████████                                                                                    | 352/800 [01:31<01:56,  3.85it/s, avr_loss=0]
epoch 23/50
steps:  46%|█████████████████████████████████████████████████████████████████████                                                                                 | 368/800 [01:35<01:52,  3.85it/s, avr_loss=0]
epoch 24/50
steps:  48%|████████████████████████████████████████████████████████████████████████                                                                              | 384/800 [01:39<01:47,  3.85it/s, avr_loss=0]
epoch 25/50
steps:  50%|███████████████████████████████████████████████████████████████████████████                                                                           | 400/800 [01:43<01:43,  3.86it/s, avr_loss=0]
epoch 26/50
steps:  52%|██████████████████████████████████████████████████████████████████████████████                                                                        | 416/800 [01:47<01:39,  3.86it/s, avr_loss=0]
epoch 27/50
steps:  54%|█████████████████████████████████████████████████████████████████████████████████                                                                     | 432/800 [01:51<01:35,  3.86it/s, avr_loss=0]
epoch 28/50
steps:  56%|████████████████████████████████████████████████████████████████████████████████████                                                                  | 448/800 [01:56<01:31,  3.86it/s, avr_loss=0]
epoch 29/50
steps:  58%|███████████████████████████████████████████████████████████████████████████████████████                                                               | 464/800 [02:00<01:26,  3.86it/s, avr_loss=0]
epoch 30/50
steps:  60%|██████████████████████████████████████████████████████████████████████████████████████████                                                            | 480/800 [02:04<01:22,  3.87it/s, avr_loss=0]
saving checkpoint: G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\models\loras\lpf-000030.safetensors

epoch 31/50
steps:  62%|█████████████████████████████████████████████████████████████████████████████████████████████                                                         | 496/800 [02:08<01:18,  3.86it/s, avr_loss=0]
epoch 32/50
steps:  64%|████████████████████████████████████████████████████████████████████████████████████████████████                                                      | 512/800 [02:12<01:14,  3.86it/s, avr_loss=0]
epoch 33/50
steps:  66%|███████████████████████████████████████████████████████████████████████████████████████████████████                                                   | 528/800 [02:16<01:10,  3.87it/s, avr_loss=0]
epoch 34/50
steps:  68%|██████████████████████████████████████████████████████████████████████████████████████████████████████                                                | 544/800 [02:20<01:06,  3.87it/s, avr_loss=0]
epoch 35/50
steps:  70%|█████████████████████████████████████████████████████████████████████████████████████████████████████████                                             | 560/800 [02:24<01:02,  3.87it/s, avr_loss=0]
epoch 36/50
steps:  72%|████████████████████████████████████████████████████████████████████████████████████████████████████████████                                          | 576/800 [02:28<00:57,  3.87it/s, avr_loss=0]
epoch 37/50
steps:  74%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████                                       | 592/800 [02:32<00:53,  3.87it/s, avr_loss=0]
epoch 38/50
steps:  76%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████                                    | 608/800 [02:37<00:49,  3.87it/s, avr_loss=0]
epoch 39/50
steps:  78%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                                 | 624/800 [02:41<00:45,  3.87it/s, avr_loss=0]
epoch 40/50
steps:  80%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                              | 640/800 [02:45<00:41,  3.87it/s, avr_loss=0]
saving checkpoint: G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\models\loras\lpf-000040.safetensors

epoch 41/50
steps:  82%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                           | 656/800 [02:49<00:37,  3.86it/s, avr_loss=0]
epoch 42/50
steps:  84%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                        | 672/800 [02:54<00:33,  3.86it/s, avr_loss=0]
epoch 43/50
steps:  86%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                     | 688/800 [02:58<00:28,  3.86it/s, avr_loss=0]
epoch 44/50
steps:  88%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                  | 704/800 [03:02<00:24,  3.87it/s, avr_loss=0]
epoch 45/50
steps:  90%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████               | 720/800 [03:06<00:20,  3.87it/s, avr_loss=0]
epoch 46/50
steps:  92%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████            | 736/800 [03:10<00:16,  3.87it/s, avr_loss=0]
epoch 47/50
steps:  94%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████         | 752/800 [03:13<00:12,  3.88it/s, avr_loss=0]
epoch 48/50
steps:  96%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████      | 768/800 [03:17<00:08,  3.88it/s, avr_loss=0]
epoch 49/50
steps:  98%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████   | 784/800 [03:21<00:04,  3.88it/s, avr_loss=0]
epoch 50/50
steps: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 800/800 [03:26<00:00,  3.88it/s, avr_loss=0]
saving checkpoint: G:\AIGeneration\ComfyUI_windows_portable\ComfyUI\models\loras\lpf.safetensors
model saved.
steps: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 800/800 [03:26<00:00,  3.88it/s, avr_loss=0]
Train finished
Prompt executed in 222.67 seconds`
